src/pairselection.hxx

```diff
@@ -387,30 +395,62 @@ auto PairSelectionAlgo() {
-            static_cast<int>(original_tau_indices[selected_tau_index])};
         Logger::get("PairSelection")
-            ->debug("Selected original pair indices: mu = {} , tau = {}",
-                    selected_pair[0], selected_pair[1]);
+            ->debug("Selected TauPt: {}", selected_tau_pt);
         Logger::get("PairSelection")
-            ->debug("MuonPt = {} , TauPt = {} ",
-                    muonpt[static_cast<UInt_t>(selected_pair[0])],
-                    taupt[static_cast<UInt_t>(selected_pair[1])]);
+            ->debug("Selected TauIso: {}", selected_tau_iso);
         Logger::get("PairSelection")
-            ->debug("MuonIso = {} , TauIso = {} ",
-                    muoniso[static_cast<UInt_t>(selected_pair[0])],
-                    tauiso[static_cast<UInt_t>(selected_pair[1])]);
+            ->debug("Selected MuonPt: {}", selected_muon_pt);
+        Logger::get("PairSelection")
+            ->debug("Selected MuonIso: {}", selected_muon_iso);
+
+        // construct the four vectors of the selected muons and taus to check
+        // deltaR and reject a pair if the candidates are too close
+
+        for (auto &candidate : sorted_pairs) {
+            auto muonindex = original_muon_indices[candidate.first];
+            ROOT::Math::PtEtaPhiMVector muon = ROOT::Math::PtEtaPhiMVector(
+                muon_pt.at(muonindex), muon_eta.at(muonindex),
+                muon_phi.at(muonindex), muon_mass.at(muonindex));
+            Logger::get("PairSelection")
+                ->debug("{} muon vector: {}", muonindex, muon);
+            auto tauindex = original_tau_indices[candidate.second];
+            ROOT::Math::PtEtaPhiMVector tau = ROOT::Math::PtEtaPhiMVector(
+                tau_pt.at(tauindex), tau_eta.at(tauindex), tau_phi.at(tauindex),
+                tau_mass.at(tauindex));
+            Logger::get("PairSelection")
+                ->debug("{} tau vector: {}", tauindex, tau);
+            Logger::get("PairSelection")
+                ->debug("DeltaR: {}",
+                        ROOT::Math::VectorUtil::DeltaR(muon, tau));
+            if (ROOT::Math::VectorUtil::DeltaR(muon, tau) > mindeltaR) {
+                Logger::get("PairSelection")
+                    ->debug(
+                        "Selected original pair indices: mu = {} , tau = {}",
+                        muonindex, tauindex);
+                Logger::get("PairSelection")
+                    ->debug("MuonPt = {} , TauPt = {} ", muon.Pt(), tau.Pt());
+                Logger::get("PairSelection")
+                    ->debug("MuonPt = {} , TauPt = {} ", muon_pt[muonindex],
+                            tau_pt[tauindex]);
+                Logger::get("PairSelection")
+                    ->debug("MuonIso = {} , TauIso = {} ", muon_iso[muonindex],
+                            tau_iso[tauindex]);
+                selected_pair = {static_cast<int>(muonindex),
+                                 static_cast<int>(tauindex)};
+                break;
```

Comment:
What happens if all `sorted_pairs` fail the deltaR criteria. Does `selected_pair` have default values?

Reply:
you are correct, this was missing !

Reply:
Should be fixed now